src/shared/dlt_config_file_parser.c

```diff
@@ -494,8 +494,8 @@ int dlt_config_file_get_section_name(const DltConfigFile *file,
     if ((file == NULL) || (name == NULL) || (num < 0) || (num >= file->num_sections))
         return -1;
 
-    strncpy(name, (file->sections + num)->name, DLT_CONFIG_FILE_ENTRY_MAX_LEN);
-
+    strncpy(name, (file->sections + num)->name, DLT_CONFIG_FILE_ENTRY_MAX_LEN-1);
```

Comment:
`-1` here is unnecessary.
`strncpy` will copy the finishing `\0` from the source buffer (here `file->sections + num)->name`) if it was there. If not, putting the '\0' at the end of the destination buffer is enough, as suggested by @ssugiura (as long as both buffers are size-compatible).